sdk/appcenter/src/main/java/com/microsoft/appcenter/http/TLS1_2SocketFactory.java

```diff
@@ -14,10 +17,30 @@
  */
 class TLS1_2SocketFactory extends SSLSocketFactory {
 
+    /**
+     * TLS 1.2 protocol name.
+     */
+    private static final String TLS1_2_PROTOCOL = "TLSv1.2";
+
     /**
      * Protocols that we allow.
      */
-    private static final String[] ENABLED_PROTOCOLS = {"TLSv1.2"};
+    private static final String[] ENABLED_PROTOCOLS = { TLS1_2_PROTOCOL };
+
+    private SSLSocketFactory mSocketFactory;
+
+    TLS1_2SocketFactory() {
+        try {
+            SSLContext sc = SSLContext.getInstance(TLS1_2_PROTOCOL);
```

Comment:
Why do we need this?

Reply:
1. #917
2. All the materials I found on the topic use this

This is the main version of #917. So opposite, I wonder why it wasn't here.
Also, I thought about lack of `ProviderInstaller.installIfNeeded` call. But this requires play-services dependency. So, it should be discussed first.